### PR TITLE
[2.7][Process] Fix memory issue when using large input streams

### PR DESCRIPTION
**Pull request by romainneutron**
| Q | A |
| --- | --- |
| Branch | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #17667 |
| License | MIT |

This removes buffering of a stream input. Tests shows that this implementation is much faster and fixes some issues when streams are based on very large inputs

### CODE REVIEW

File: src/Symfony/Component/Process/Pipes/UnixPipes.php [hunk 1]
```diff
@@ -147,16 +149,16 @@ public function readAndWrite($blocking, $close = false)
...
-                if ($type === 'input') {
-                    $this->inputBuffer .= $data;
-                } else {
-                    $read[$type] = $data;
+            if ($type !== 'input') {
+                while ('' !== $dataread = (string) fread($pipe, self::CHUNK_SIZE)) {
+                    $data .= $dataread;
+                }
+                // Remove extra null chars returned by fread
+                if ($data !== '') {
```
Comment (javiereguiluz) [inline src/Symfony/Component/Process/Pipes/UnixPipes.php]: Replace it by `if ('' !== $data) {` for consistency?

**Comment by romainneutron** [comment]
Tests are okay on component process on both CI

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @romainneutron.

**Pull request merged**